src/transformers/integrations/integration_utils.py

```diff
@@ -246,19 +248,28 @@ def _objective(trial, local_trainer, checkpoint_dir=None):
         except ModuleNotFoundError:
             pass
 
-        checkpoint = None
-        if checkpoint_dir:
-            for subdir in os.listdir(checkpoint_dir):
-                if subdir.startswith(PREFIX_CHECKPOINT_DIR):
-                    checkpoint = os.path.join(checkpoint_dir, subdir)
+        checkpoint = ray.train.get_checkpoint()
+        checkpoint_path = None
+        if checkpoint:
+            with checkpoint.as_directory() as checkpoint_dir:
+                for subdir in os.listdir(checkpoint_dir):
+                    if subdir.startswith(PREFIX_CHECKPOINT_DIR):
+                        checkpoint_path = os.path.join(checkpoint_dir, subdir)
```

Comment:
Doesn't the `checkpoint_dir` get removed after getting out of scope?

I think we need `checkpoint.to_directory()` here instead

Reply:
Good catch, fixed and tested restoration with an updated test (in the ray repo which I'll link)